core/modules/qdisp/JobStatus.cc

```diff
@@ -0,0 +1,162 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2015 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+ /**
+  * @file
+  *
+  * @brief JobStatus implementation
+  *
+  * Store information message issused by a job
+  * running a chunkQuery on a xrootd ressource
+  *
+  * @author Fabrice Jammes, IN2P3
+  */
+
+// Class header
+#include "qdisp/JobStatus.h"
+
+// System headers
+#include <ctime>
+#include <iostream>
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+namespace lsst {
+namespace qserv {
+namespace qdisp {
+
+namespace {
+
+
+LOG_LOGGER getLogger() {
+    static LOG_LOGGER logger = LOG_GET("lsst.qserv.qdisp.JobStatus");
+    return logger;
+}
+
+}
+
+// Static fields
+std::string const JobStatus::_empty;
+
+JobStatus::Info::Info(ResourceUnit const& resourceUnit_)
+    : resourceUnit(resourceUnit_),
+      state(UNKNOWN),
+      stateCode(0) {
+    stateTime = ::time(NULL);
+}
+
+void JobStatus::updateInfo(JobStatus::State s, int code /* =0 */, std::string const& desc /* =_empty */) {
+    std::lock_guard<std::mutex> lock(_mutex);
+
+    LOGF(getLogger(), LOG_LVL_TRACE, "Updating %1% state to: %2%" % (void*)this % s);
+    _info.stateTime = ::time(NULL);
+    _info.state = s;
+    _info.stateCode = code;
+    _info.stateDesc = desc;
+}
+
+std::ostream& operator<<(std::ostream& os, JobStatus::State const& state) {
+    switch(state)
+    {
+    case JobStatus::UNKNOWN:
+        os << "Unknown";
```

Comment:
This one is probably OK; only one call will be executed, and the space costs are trivial.